### PR TITLE
Feature/codegen refactor

### PR DESCRIPTION
**Pull request by ekampf**

### CODE REVIEW

File: gql/utils_codegen.py [hunk 1]
```diff
@@ -4,6 +4,17 @@
 
 
 class CodeGenerator:
```
Comment (aviv-ebates) [inline gql/utils_codegen.py]: This class should not be called `CodeGenerator`. Maybe `CodeChunk`.

File: gql/utils_codegen.py [hunk 2]
```diff
@@ -30,6 +41,13 @@ def write_lines(self, lines):
         for line in lines:
             self.lines.append(self.indent_string + line)
 
+    def block(self):
+        return self.Block(self)
+
+    def write_block(self, block_header: str, *args, **kwargs):
```
Comment (aviv-ebates) [inline gql/utils_codegen.py:47]: I'm also not happy about this name either.

File: gql/utils_codegen.py [hunk 3]
```diff
@@ -4,6 +4,17 @@
...
+    class Block:
+        def __init__(self, codegen: 'CodeGenerator'):
+            self.gen = codegen
+
+        def __enter__(self):
+            self.gen.indent()
+            return self.gen
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            self.gen.undent()
```
Comment (aviv-ebates) [inline gql/utils_codegen.py]: `unindent`

File: gql/utils_codegen.py (hunk 2 above)
Comment (ekampf) [reply gql/utils_codegen.py]: `begin_block` ?

Comment (ekampf) [reply gql/utils_codegen.py:47]: @aviv-ebates maybe `begin_block` ?

Comment (ekampf) [reply gql/utils_codegen.py:47]: no begin is not good because it also in charge of ending...

**Pull request merged**